include/af/sparse.h

```diff
@@ -0,0 +1,85 @@
+/*******************************************************
+ * Copyright (c) 2014, ArrayFire
+ * All rights reserved.
+ *
+ * This file is distributed under 3-clause BSD license.
+ * The complete license agreement can be obtained at:
+ * http://arrayfire.com/licenses/BSD-3-Clause
+ ********************************************************/
+
+#pragma once
+#include <af/defines.h>
+
+#ifdef __cplusplus
+namespace af
+{
+    class array;
+
+    AFAPI array createSparseArray(const dim_t nRows, const dim_t nCols, const dim_t nNZ,
+                                  const array values, const array rowIdx, const array colIdx,
+                                  const af::sparseStorage storage = AF_SPARSE_CSR);
+
+    AFAPI array createSparseArray(const dim_t nRows, const dim_t nCols, const dim_t nNZ,
+                                  const void* const values,
+                                  const int * const rowIdx, const int * const colIdx,
+                                  const dtype type = f32, const af::sparseStorage storage = AF_SPARSE_CSR,
+                                  const af::source src = afHost);
+
+    AFAPI array createSparseArray(const array dense, const af::sparseStorage storage = AF_SPARSE_CSR);
+
+    AFAPI array sparseConvertStorage(const array in, const af::sparseStorage storage);
+
+    AFAPI void sparseGetArrays(array &values, array &rowIdx, array &colIdx, const array in);
+
+    AFAPI array sparseGetValues(const array in);
+
+    AFAPI array sparseGetRowIdx(const array in);
+
+    AFAPI array sparseGetColIdx(const array in);
+
+    AFAPI dim_t sparseGetNumNonZero(const array in);
+
+    AFAPI af::sparseStorage sparseGetStorage(const array in);
```

Comment:
@umar456 I asked him to make these functions on a whim, but I reconsidering making them as methods to `af::array` class.

Another alternative to naming these functions would be to use something like `af::sparse::array(...)` similar to `afcl::array` we have in `opencl.h`